docs/proposals/PROPOSAL_D.md

```diff
@@ -0,0 +1,413 @@
+- A **Namespace** is a named identifies that provides an organization under which users can specify compatibility objects.
+  - Owned and registered to a specific CIG
+
+The new image compatibility specification includes a JSON compatibility structure with keys that are owned by compatibility interest groups, and include simple key, value pairs (nodes) and optionally relationships (edges) that can be harnassed by downstream tools for analysis of compatibility. Importantly, there is no complex graph structure in artifact itself, but rather in the single, top level schema that defines the space of metadata attributes and relationships. The tools can use the graph (nodes and edges) as needed for their use case. This gives power and freedom to tool developers to create tools as they see fit. Several designs are afforded:
+
+1. **Simple checking of key/value pairs**: A community might have an algorithm that is as simple as their tool checking for particular keys, and that values match for a host of interest.
+2. **Graph-based logic**: A community might warrant using advanced graph parsing to do a similar assessment. An example comes from [Archspec](https://github.com/archspec), where a single architecture (from a flat artifact or the platform, depending on what the tool chooses) can [be placed in a graph](https://github.com/archspec/archspec-json/blob/master/cpu/microarchitectures.json) and used for advanced assessment of compatibility.
+
+The above design encourages simplicity and designation of fewer metadata pieces in the OCI objects. Any complex graph-based query or additional relationships should be represented on the level of the tools that do the compatibility checking. This ensures that changes to these methods or graphs do not require re-creating extensive numbers of artifacts.
+
+For the above, the assumption is that there are two kinds of metadata:
+
+- **bag of features** (nodes without edges) that might go into a simple (even brute force) matching algorithm.
+- **graph** (nodes with edges) relationships that can be used by a higher level tool to more optimially prune the graph before the detailed annotations are considered.
+
+It is up to the interest group if they want to define the additional relationships (edges). If their features have any kind of relationship that might be considered in a graph analysis, the graph adds additional functionality for that. The tools can choose to use the graph (edges and nodes) as needed. Arguably, the edges could be ignored and only the metadata attributes (nodes) considered.
+
+## Modifications
+
+The compatibility schema that defines the structure of the artifacts and relationships between attributes would need to live in a single, known location, a web accessible URL that would be maintained by the compatibility interest group, and would be versioned. The artifacts would reference the version they are using. The compatibility artifact JSON blob would live alongside images in a registry.
```

Comment:
That's interesting idea to add schema per intereset group. However, it cannot always be downloaded from the internet. We have to support secure and air-gapped environments.

Reply:
Good point - this is something we deal with in HPC too. It would need to be cached in advance. There is the additional problem that without internet, they couldn't pull a container image to begin with!

Reply:
I would argue the tool that does the original generation of the compressed package should include compatibility artifacts. That isn't an issue specific to this proposal in particular, but to any proposal that has the metadata separate from the manifest.